common/persistence/visibility/visibility_manager_impl.go

```diff
@@ -247,6 +247,24 @@ func (p *visibilityManagerImpl) CountWorkflowExecutions(
 	return response, err
 }
 
+func (p *visibilityManagerImpl) GetWorkflowExecution(
+	ctx context.Context,
+	request *manager.GetWorkflowExecutionRequest,
+) (*manager.GetWorkflowExecutionResponse, error) {
+	response, err := p.store.GetWorkflowExecution(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+	if response == nil {
+		return nil, nil
+	}
```

Comment:
And then you don't need this.

Reply:
Actually you don't need anyway. If `err == nil`, then `response` can't be `nil`.

Reply:
I added this because of unit test. One of the mocks returns nil.

Reply:
Removed this, and changed unit test mock to return a dummy response.